File: U1/EXAMEN/02 CLICON/Comer_CliCon_SOAP_DotNet_GR01/Comer_CliCon_SOAP_DotNet_GR01/Models/SolicitudCalculoDTO.cs
using System.Runtime.Serialization;

namespace Comer_CliCon_SOAP_DotNet_GR01.Models
{
    /// <summary>
    /// DTO específico para calcular total de factura
    /// Solo requiere los productos (Items)
    /// </summary>
    [DataContract(Namespace = "http://schemas.datacontract.org/2004/07/Comercializadora_Soap_DotNet_GR01.DTOs")]
    public class SolicitudCalculoDTO
    {
        [DataMember(IsRequired = true, Order = 1)]
        public List<ItemFacturaDTO> Items { get; set; } = new();
    }
}

File: U1/PROYECTO/01 SERVIDOR/Comercializadora_Restful_Java_GR01/src/main/java/ec/edu/monster/dto/SolicitudCalculoDTO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.dto;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/**
 * DTO específico para calcular total de factura
 * Solo requiere los productos (Items)
 */
public class SolicitudCalculoDTO implements Serializable {

    private List<ItemFacturaDTO> items = new ArrayList<>();

    // Constructores
    public SolicitudCalculoDTO() {
    }

    public SolicitudCalculoDTO(List<ItemFacturaDTO> items) {
        this.items = items;
    }

    // Getters y Setters
    public List<ItemFacturaDTO> getItems() {
        return items;
    }

    public void setItems(List<ItemFacturaDTO> items) {
        this.items = items;
    }
}
